### ISSUE TITLE
Formatting Breaks `add_library` Call

### ISSUE THREAD

**Issue by sanssecours** [cheshirekow/cmake_format#111]
# Steps to Reproduce

1. Create a file `test.cmake` with the following content:

   ```cmake
   add_library (cframework OBJECT ${SOURCES})
   ```

3. Run `cmake-format`:

   ```sh
   cmake-format test.cmake
   ```

# Expected Result

The command `cmake-format` prints a nicely formatted version of the given CMake code.

# Actual Result

cmake format breaks the `add_library` function call, since it moves the keyword `OBJECT`:

```cmake
add_library(cframework ${SOURCES} OBJECT)
```

.

# System Information

- cmake format `0.5.1`
- macOS `10.14.4`

**Comment by cheshirekow** [comment]
You can disable the `autosort` config as a work-around.

**Comment by cheshirekow** [comment]
Ok, I added parsers for the non-"normal" `add_library()` forms. They are in the branch `stage_v0.5.2` ([here][1]) if you'd like to try. I definitely should have done that before defaulting to `autosort=on`. I forgot that there were additional forms of this command. I'll need to add additional forms for `add_executable()` too. 

[1]: https://github.com/cheshirekow/cmake_format/tree/stage_v0.5.2

**Comment by cheshirekow** [comment]
Also, as a note to myself: autosort will also break if the command form descriminator is hidden behind a cmake variable. For example:

```
set(libtype OBJECT)
add_library(foobar ${libtype} bar.cc baz.cc foo.cc)
```

**Comment by gocarlos** [comment]
same happens for ALIAS:
```cmake
# correct
add_library(my_proj::my_comp
            ALIAS 
            ${PROJECT_NAME}_my_comp
            )
# not correct
add_library(my_proj::my_comp
            ${PROJECT_NAME}_my_comp
            ALIAS)
```

**Comment by cheshirekow** [comment]
@gocarlos I believe this case is already covered in `stage_v0.5.2` ([here][1]) but if you are seeing this on that branch please let me know:

[1]: https://github.com/cheshirekow/cmake_format/tree/stage_v0.5.2

**Comment by cheshirekow** [comment]
Closed in v0.5.2
